projects/ngx-indexed-db/src/lib/ngx-indexed-db.service.ts

```diff
@@ -253,16 +253,16 @@ export class NgxIndexedDBService {
 		});
 	}
 
-	getByIndex(storeName: string, indexName: string, key: any) {
-		return new Promise<any>((resolve, reject) => {
+	getByIndex<T>(storeName: string, indexName: string, key: any) {
+		return new Promise<T>((resolve, reject) => {
 			openDatabase(this.indexedDB, this.dbConfig.name, this.dbConfig.version).then(db => {
 				validateBeforeTransaction(db, storeName, reject);
 				let transaction = createTransaction(db, optionsGenerator(DBMode.readonly, storeName, reject, resolve)),
 					objectStore = transaction.objectStore(storeName),
 					index = objectStore.index(indexName),
 					request = index.get(key);
 				request.onsuccess = (event: Event) => {
-					resolve((<IDBOpenDBRequest>event.target).result);
+					resolve((<any>event.target).result);
```

Comment:
Why any here?

Reply:
A couple reasons, though none of them are great 😅 :
1. Without it, you get a type error:
    ![image](https://user-images.githubusercontent.com/305254/89559403-784e8980-d7ca-11ea-9964-6c3a5ac6974e.png)
1. Other methods do something similar, so I figured it was okay to copy that pattern here:
    - https://github.com/assuncaocharles/ngx-indexed-db/blob/35fcb751b1f94f3333c812deaba51dd6f12a5160/projects/ngx-indexed-db/src/lib/ngx-indexed-db.service.ts#L72
    - https://github.com/assuncaocharles/ngx-indexed-db/blob/35fcb751b1f94f3333c812deaba51dd6f12a5160/projects/ngx-indexed-db/src/lib/ngx-indexed-db.service.ts#L90

Reply:
1. It's only wrong typing, the proper solution should be:

```typescript

	getByKey<T>(storeName: string, key: any) {
		return new Promise<T>((resolve, reject) => {
			openDatabase(this.indexedDB, this.dbConfig.name, this.dbConfig.version)
				.then((db: IDBDatabase) => {
					let transaction = createTransaction(
							db,
							optionsGenerator(DBMode.readonly, storeName, reject, resolve)
						),
						objectStore = transaction.objectStore(storeName);
					let request = objectStore.get(key) as IDBRequest<T>;
					request.onsuccess = function(event) {
						resolve((event.target as IDBRequest<T>).result);
					};
					request.onerror = function(event) {
						reject(event);
					};
				})
				.catch(reason => reject(reason));
		});
	}
```

2. The idea is always to reduce/eliminate the `any` type

Reply:
Ah okay, I wasn't sure what to use instead of `any`. Thank you for that! I've replaced a couple `any`s with `IDBRequest<T>`.